modules/system/resource-group/docs/features/0003-entity-hierarchy.md

```
@@ -0,0 +1,1374 @@
+  - [Unit Test Coverage for Entity Hierarchy](#unit-test-coverage-for-entity-hierarchy)
+  - [REST API Test Coverage](#rest-api-test-coverage)
+- [6. Acceptance Criteria](#6-acceptance-criteria)
+- [7. Unit Test Plan](#7-unit-test-plan)
+  - [Module-Level Summary](#module-level-summary)
+  - [Coverage Summary](#coverage-summary)
+  - [Entity Hierarchy Test Cases](#entity-hierarchy-test-cases)
+  - [Group `metadata` — Barrier as Data](#group-metadata--barrier-as-data)
+  - [REST-level metadata serialization](#rest-level-metadata-serialization)
+  - [Invalid / Non-GTS Input Tests](#invalid--non-gts-input-tests)
+  - [ADR-001 Hierarchy Reproduction in RG Module](#adr-001-hierarchy-reproduction-in-rg-module)
+  - [metadata Validation Against Type's metadata_schema](#metadata-validation-against-types-metadataschema)
```

markdownlint-cli2 (0.22.0)

[warning] Link fragments should be valid

(MD051, link-fragments)